src/Features/LanguageServer/Protocol/LspMiscellaneousFilesWorkspace.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Features.Workspaces;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.LanguageServer
+{
+    /// <summary>
+    /// Defines a default workspace for opened LSP files that are not found in any
+    /// workspace registered by the <see cref="ILspWorkspaceRegistrationService"/>.
+    /// If a document added here is subsequently found in a registered workspace, 
+    /// the document is removed from this workspace.
+    /// 
+    /// Future work for this workspace includes supporting basic metadata references (mscorlib, System dlls, etc),
+    /// but that is dependent on having a x-plat mechanism for retrieving those references from the framework / sdk.
+    /// </summary>
+    internal class LspMiscellaneousFilesWorkspace : Workspace
+    {
+        private const string LspMiscellaneousFilesWorkspaceKind = nameof(LspMiscellaneousFilesWorkspaceKind);
```

Comment:
Can this reuse the existing MiscellaneousWorkspace kind? Some features do check if they're in a misc files workspace to disable themselves; I would imagine we want that to still work. We'd have to confirm we don't have services exporting on it though.